runtime/src/bpf_map/userspace/lpm_trie_map.hpp

```diff
@@ -0,0 +1,133 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Copyright (c) 2024, bpftime contributors
+ * All rights reserved.
+ */
+#ifndef _LPM_TRIE_MAP_HPP
+#define _LPM_TRIE_MAP_HPP
+
+#include <boost/interprocess/managed_shared_memory.hpp>
+#include <boost/interprocess/allocators/allocator.hpp>
+#include <boost/interprocess/containers/vector.hpp>
+#include <memory>
+#include <cstdint>
+
+namespace bpftime
+{
+
+// LPM Trie key structure (matches kernel's bpf_lpm_trie_key)
+struct bpf_lpm_trie_key {
+	uint32_t prefixlen; // Prefix length in bits
+	uint8_t data[]; // Data in network byte order (big endian)
+};
+
+// Forward declaration
+struct lpm_trie_node;
+
+// LPM Trie node structure
+struct lpm_trie_node {
+	std::shared_ptr<lpm_trie_node> child[2]; // Left and right children
```

Comment:
Use smart pointers from boost::interprocess, see https://www.boost.org/doc/libs/latest/doc/html/interprocess/interprocess_smart_ptr.html#interprocess.interprocess_smart_ptr.shared_ptr

Using std smart pointers may lead to segmentation fault in shared memory